src/librustc/hir/map/mod.rs

```diff
@@ -497,15 +497,15 @@ impl<'hir> Map<'hir> {
     }
 
     pub fn trait_impls(&self, trait_did: DefId) -> &'hir [NodeId] {
-        self.dep_graph.read(DepNode::TraitImpls(trait_did));
+        self.dep_graph.read(DepNode::AllLocalTraitImpls);
```

Comment:
I'm curious why you converted this into a single dep-node, rather than keeping it per-trait. It seems like this is losing quite a bit of precision, no? 

(That is, if **any** trait adds an impl, we will consider them **all** to have changed.)